### PR TITLE
Library translation registering

### PR DESCRIPTION
**Pull request by jkimbo**
First stab at implementing an `register_domain` method that libraries can use to register their translations.

### CODE REVIEW

File: kano_i18n/init.py [hunk 1]
```diff
@@ -9,8 +9,50 @@
...
+    """
+    Register a domain as a fallback to the currently installed translation.
+    Used by library modules to register their translations on top of the current
+    app translation.
+    """
+    global REGISTERED_DOMAINS
+    global CURRENT_TRANSLATION
+
+    if not hasattr(__builtin__, '_') or not CURRENT_TRANSLATION:
+        # TODO
```
Comment (jkimbo) [inline kano_i18n/init.py]: Unsure what the best thing to do here would be. Any suggestions @Ealdwulf @tombettany ?

File: kano_i18n/init.py [hunk 2]
```diff
@@ -9,8 +9,50 @@
...
+        return
+
+    # Create translation
+    trans = gettext.translation(
+        domain,
+        localedir=locale_dir,
+        fallback=True  # don't throw an error if the no .mo files found for domain, just return `NullTranslations`
+    )
+
+    CURRENT_TRANSLATION.add_fallback(trans)
```
Comment (Ealdwulf) [inline kano_i18n/init.py]: Are you sure you can add multiple fallbacks to a translation?I thought they had to form a chain.

Comment (jkimbo) [inline kano_i18n/init.py]: Yes you can. The relevant snippet it is:

```python
    def add_fallback(self, fallback):
        if self._fallback:
            self._fallback.add_fallback(fallback)
        else:
            self._fallback = fallback
```
https://hg.python.org/cpython/file/2.7/Lib/gettext.py#l187

Comment (Ealdwulf) [inline kano_i18n/init.py]: 👍

**Comment by jkimbo** [comment]
@Ealdwulf added the deferred registering logic

**Comment by Ealdwulf** [comment]
Looks good. When you feel it's tested sufficiently, feel free to merge.

**Comment by jkimbo** [comment]
Thanks @Ealdwulf Do I have to bump the version as well?

**Comment by Ealdwulf** [comment]
@jkimbo Yes, thanks for remembering...

**Pull request merged**